src/serialization/UntypedLong.cs

```diff
@@ -0,0 +1,18 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Kiota.Abstractions.Serialization
+{
+    /// <summary>
+    /// Represents an untyped node with long value.
+    /// </summary>
+    /// <param name="value">The long value associated with the node.</param>
+    public class UntypedLong(long value) : UntypedNode
+    {
+        /// <summary>
+        /// The value associated with untyped long node.
+        /// </summary>
+        public new long Value { get; } = value;
```

Comment:
instead of hiding the base property, which is not going to work in most other languages, could we:
1. move most of what's in "UntypedNode" to an abstract generic "BaseUntypedNode" class.
2. make all classes that derive from "UntypedNode" derive from "BaseUntypedNode" and pass the generic type parameter.
3. Remove the constructors, or add default constructors (parameterless)
4. make "UntypedNode" derive from "BaseUntypedNode", and pass object as the type.

throughts?

Reply:
Can we remove UntypedNode? If all Untyped classes will derive from BaseUntypedNode?
Not sure about UntypedObject which has properties (untyped nodes but with different type)
maybe
`UntypedObject : BaseUntypedNode<IDictionary<string, UntypedNode>>`

Reply:
it went through my mind as well, I wasn't sure "how much we needed it". It's worth considering we might need it in other languages because of type erasure and similar concepts.

Reply:
In my head, the need for `UntypedNode` is really for a type to generate in the code that can encompass all types(by being the base) so that it can be evaluated at runtime. 
If `UntypedNode` is no longer a base of the rest and `BaseUntypedNode` is, we would end up generating `BaseUntypedNode` and there may be no need for `UntypedNode` because for handling the "object" case, we already have `UntypedObject ` for that.

The use of generics here at the base may lead to type conversion challenges at runtime given that the generic parameter needed is unknown at compile/generation time. E.g. if the base is declared as `BaseUntypedNode<T>` a method trying to pull out `UntypedObject` and `UntypedBoolean` what would you generate as `T`? (If we generate `BaseUntypedNode<object>`, this would be like converting a `List<bool>` to a `List<object>`.

Given @baywet points, maybe we could have this property as a method intead as `GetValue()` and the return type based on the node type and keep the structure as is? That way we can use function/method semantics similar to what we do in other generation scenarios across languages.

Reply:
Yes. I think that can work and method overriding with covariant return types(different return types in derived classes inheriting from object should still be consistently achievable in other languages).